editor.planx.uk/src/@planx/components/PlanningConstraints/utils.ts

```diff
@@ -0,0 +1,69 @@
+import { Constraint } from "@opensystemslab/planx-core/types";
+import { InaccurateConstraints } from "./Public";
+import { IntersectingConstraints } from "./model";
+
+export function handleOverrides(
```

Comment:
This is a pretty dense function I had a bit of a hard time following (clearly time for another coffee! ☕). Unit tests are helpful and clear however 👍 

There's maybe a few minor things that could be done here, like splitting up into smaller functions or guard clauses etc that might clear things up, but I'm not sure how much overall difference this would make.

I wonder if a significant change could be just to update the framing slightly. Rather than filtering from intersecting and pushing to nots, I wonder if it might be clearer if we were constructing a new object of intersecting constraints? For example - 
 - For each constraint key and value...
 - If not inaccurate, push to new constraint list
 - If inaccurate, push to nots
 
This might not be a viable strategy, and is maybe just totally personal preference, but thought it was worth sharing 👍

Reply:
Totally agree this is a dense one - re-organised a bit and I think it feels marginally better ? Very much counting on the tests to do the hard work here :sweat_smile: 

I'm using an approach of iterating through each inaccuracy, rather than each constraint, for a couple of reasons:
- The average case probably won't have inaccuracies, so this will exit immediately
- For cases that do have inaccuracies, the hard part of checking is reconciling the entities within the overall constraint categories and updating parent (eg `designated`) or child (eg `listed.grade.II`) keys correctly - I think this is a bit more straightforward to think about when starting from a single entity ID that we know is inaccurate, rather than starting from the constraint category that has zero to many entities which we then have to check if they are inaccurate or not - but that may very well just be my preference for thinking about this lately !

Reply:
Thanks for taking another look, agree this is better! And yep - the tests are fantastic here which is massively helpful 😄